.github/workflows/end2end-test-playwright.yml

```
@@ -0,0 +1,68 @@
+name: End-to-End Tests Playwright
+
+on:
+    pull_request:
+    push:
+        branches:
+            - trunk
+            - 'release/**'
+            - 'wp/**'
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+    # The concurrency group contains the workflow name and the branch name for pull requests
+    # or the commit hash for any other events.
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    e2e:
+        name: E2E Tests
+        runs-on: ubuntu-latest
+        if: ${{ github.repository == 'WordPress/gutenberg' || github.event_name == 'pull_request' }}
+        strategy:
+            fail-fast: false
+            matrix:
+                node: ['14']
+
+        steps:
+            - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+
+            - name: Use desired version of NodeJS
+              uses: actions/setup-node@38d90ce44d5275ad62cc48384b3d8a58c500bb5f # v2.2.2
+              with:
+                  node-version: ${{ matrix.node }}
+                  cache: npm
+
+            - name: Npm install and build
+              run: |
+                  npm ci
+                  npm run build
+
+            - name: Install Playwright dependencies
+              run: |
+                  npx playwright install
```

```suggestion
                  npx playwright install --with-deps
```

if you are interested to run other browsers, you also need this one. Or for just Chromium you can do:

`npx playwright install chromium --with-deps`

---

What's the difference between this and `npx playwright install` or [`npx playwright install-deps`](https://playwright.dev/docs/cli#install-system-dependencies)? I'm guessing that `install` only installs the browser, `install-deps` only installs the system dependencies, and `--with-deps` does both?

Is this only necessary in CI, or should we include this in `pre-test-e2e:playwright` as well?

---

Exactly. install installs browsers, install-deps installs OS dependencies and --with-deps does both. You can specify a browser (see above) then only the necessary dependencies get installed.

Usually you can't guarantee that the GitHub Action runner stays compatible with Chromium / contains always the required dependencies, thats why we recommend it. Also e.g. when we talk about fonts / screenshot diffing.

Depends, I'd not include this inside any NPM scripts, since --with-deps requires sudo and modifies the system. When launching a Playwright browser it will yell at you, if browsers or OS dependencies are missing.